### PR TITLE
sort results in find() method

### LINKED ISSUE #2425: Question: Exclude via css path fetching?
Sometimes there are div's or paragraphs containing ads  or some other unwanted content in a css path fetched article. Is there some way to exclude/filter this content?

for example:
```
<div id='article'>
<p class='content'>wanted content</p>
<p class='add'>unwanted content</p>
<p class='content'>wanted content</p>
</div>
```
I want to have all .content, but not .add

### PR DESCRIPTION
**Pull request by printfuck**
Resolves #2425

How to test the feature manually:

My [Comment](https://github.com/FreshRSS/FreshRSS/issues/2425#issuecomment-558350468) for the issue describes it very well.

Pull request checklist:

- [X] clear commit messages
- [X] code manually tested
- [ ] unit tests written (optional if too hard)
- [ ] documentation updated

### CODE REVIEW

File: lib/lib_phpQuery.php [hunk 1]
```diff
@@ -1796,6 +1796,15 @@ protected function runQuery($XQuery, $selector = null, $compare = null) {
...
 		$this->elements = $stack;
 	}
+
+	/**
+	* used for ordering results in find()
+	*/
+	private function cmpByLineNo($a , $b ) {
+    	return ( $a->getLineNo() <= $b->getLineNo() ) ? -1 : 1;
+  	}
+
```
Comment (Alkarex) [inline lib/lib_phpQuery.php]:
```suggestion
	/**
	* used for ordering results in find()
	*/
	private function cmpByLineNo($a , $b) {
		return $a->getLineNo() <= $b->getLineNo() ? -1 : 1;
	}
```

**Comment by Alkarex** [comment]
Thanks @printfuck , that looks good :-)
Do you have an example of feed and selector to test?

File: lib/lib_phpQuery.php [hunk 2]
```diff
@@ -1958,6 +1967,7 @@ public function find($selectors, $context = null, $noHistory = false) {
 				if (! $this->elementsContainsNode($node, $stack))
 					$stack[] = $node;
 		}
+		usort($stack, array('phpQueryObject','cmpByLineNo'));
```
Comment (Alkarex) [inline lib/lib_phpQuery.php]:
```suggestion
		usort($stack, array('phpQueryObject', 'cmpByLineNo'));
```

**Comment by printfuck** [comment]
> 
> 
> Thanks @printfuck , that looks good :-)
> Do you have an example of feed and selector to test?

You can try the following expression:
```
.cluster-header p, .authors, .cluster-header figure, .formatted p, .formatted h3, #list-jtoc.listpages
```
on ``` https://rss.golem.de/rss.php?feed=RSS2.0 ```

You should see the difference in the header position.

**Comment by Alkarex** [comment]
Thanks, it seems to work as expected 👍

**Comment by Alkarex** [comment]
Could you please add a line for you in https://github.com/FreshRSS/FreshRSS/blob/master/CREDITS.md ?

**Pull request merged**